### PR TITLE
Add backward gradient for background

### PR DESCRIPTION
**Pull request by KevinXu02**
Add backward gradient for background
(related with https://github.com/nerfstudio-project/nerfstudio/pull/3100)

### CODE REVIEW

File: gsplat/rasterize.py [hunk 1]
```diff
@@ -117,10 +115,7 @@ def forward(
         num_intersects, cum_tiles_hit = compute_cumulative_intersects(num_tiles_hit)
 
         if num_intersects < 1:
-            out_img = (
-                torch.ones(img_height, img_width, colors.shape[-1], device=xys.device)
-                * background
-            )
+            out_img = torch.ones(img_height, img_width, colors.shape[-1], device=xys.device) * background
```
Comment (jb-ye) [inline gsplat/rasterize.py:118]: Should we change the annotation of 
`background: Optional[Float[Tensor, "channels"]] = None` to `background: Float[Tensor, "channels"]`. The implementation seems will rather if background is None.

File: gsplat/rasterize.py [hunk 2]
```diff
@@ -230,7 +225,8 @@ def backward(ctx, v_out_img, v_out_alpha=None):
                 v_out_img,
                 v_out_alpha,
             )
-
+        v_background = torch.matmul(v_out_img.float().view(-1, 3).t(), final_Ts.float().view(-1, 1)).squeeze()
```
Comment (jb-ye) [inline gsplat/rasterize.py:228]: Should we have a check that (Because I believe v_background could be some overhead at rendering time)
```
v_background = None
if background. requires_grad:
    v_background = ...
```

Comment (KevinXu02) [reply gsplat/rasterize.py]: thanks for the advice. fixed

**Review by jb-ye — APPROVED** [review]
